### PR TITLE
Added debug flag to GCSReader/Writer

### PR DESCRIPTION
**Pull request by Yash9060**
Added debug statement from GCSReader/ GCSWriter.

Will update documentation in follow up PR.

### CODE REVIEW

File: dataflux_pytorch/lightning/gcs_filesystem.py [hunk 1]
```diff
@@ -4,44 +4,52 @@
...
         blob = self.storage_client.bucket(bucket).blob(path)
         if mode == "wb":  # write mode.
+            if self.debug:
+                print(
+                    f'Writing from Rank: {dist.get_rank()} to bucket {bucket}\
+                        &path {path} ...')
             with DatafluxCheckpointBuffer(blob) as stream:
                 yield stream
         elif mode == "rb":  # read mode.
+            print(f'Reading on Rank: {dist.get_rank()} from bucket {bucket} &\
```
Comment (MattIrv) [inline dataflux_pytorch/lightning/gcs_filesystem.py:43]: `if self.debug:`?

**Review by MattIrv — APPROVED** [review]

**Review by jdnurme — APPROVED** [review]

**Comment by jdnurme** [comment]
I think tests need to be updated with the new flag defaulted.

File: dataflux_pytorch/lightning/gcs_filesystem.py (hunk 1 above)
Comment (Yash9060) [reply dataflux_pytorch/lightning/gcs_filesystem.py]: Done

File: dataflux_pytorch/lightning/gcs_filesystem.py [hunk 2]
```diff
@@ -4,36 +4,44 @@
...
         user_agent.add_dataflux_user_agent(self.storage_client)
+        self.debug = debug
 
     @contextmanager
     def create_stream(self, path: Union[str, os.PathLike],
                       mode: str) -> Generator[io.IOBase, None, None]:
         bucket, path = parse_gcs_path(path)
         blob = self.storage_client.bucket(bucket).blob(path)
         if mode == "wb":  # write mode.
+            print(
```
Comment (jdnurme) [inline dataflux_pytorch/lightning/gcs_filesystem.py:36]: Are these statements not intended to be behind the debug flag?

**Review by jdnurme — APPROVED** [review]

File: dataflux_pytorch/tests/test_dataflux_lightning_gcs_filesystem.py [hunk 3]
```diff
@@ -16,7 +17,9 @@ def setUp(self):
         self.bucket = fake_gcs.Bucket("fake_bucket")
         self.client = fake_gcs.Client()
         self.fake_gcs = GCSFileSystem(project_name=self.project_name,
+                                      debug=False,
                                       storage_client=self.client)
+        dist.init_process_group("gloo")
```
Comment (jdnurme) [inline dataflux_pytorch/tests/test_dataflux_lightning_gcs_filesystem.py:22]: Seems like testing doesn't like this being spun up in the fake implementation. I would either gate the printouts behind the debug flag (and pass false for testing) or I would mock `dist.get_rank()` out of all tests where it causes a failure.

**Review by jdnurme — APPROVED** [review]

File: dataflux_pytorch/tests/test_dataflux_lightning_gcs_filesystem.py (hunk 3 above)
Comment (Yash9060) [reply dataflux_pytorch/tests/test_dataflux_lightning_gcs_filesystem.py]: Removed

File: dataflux_pytorch/lightning/gcs_filesystem.py (hunk 2 above)
Comment (Yash9060) [reply dataflux_pytorch/lightning/gcs_filesystem.py]: Added debug flag

**Pull request merged**